sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs

```diff
@@ -14,20 +14,28 @@ namespace Azure.Core
 {
     internal abstract class OperationInternalBase
     {
+        private static readonly OperationPollingStrategy DefaultPollingStrategy = new ConstantPollingStrategy();
+
         private readonly ClientDiagnostics _diagnostics;
         private readonly string _updateStatusScopeName;
         private readonly IReadOnlyDictionary<string, string>? _scopeAttributes;
-        private const string RetryAfterHeaderName = "Retry-After";
-        private const string RetryAfterMsHeaderName = "retry-after-ms";
-        private const string XRetryAfterMsHeaderName = "x-ms-retry-after-ms";
 
-        protected OperationInternalBase(ClientDiagnostics clientDiagnostics, Response rawResponse, string operationTypeName, IEnumerable<KeyValuePair<string, string>>? scopeAttributes = null)
+        protected OperationInternalBase(ClientDiagnostics clientDiagnostics, Response rawResponse, string operationTypeName, IEnumerable<KeyValuePair<string, string>>? scopeAttributes = null, OperationPollingStrategy? defaultPollingStrategy = null)
         {
             _diagnostics = clientDiagnostics;
             _updateStatusScopeName = $"{operationTypeName}.UpdateStatus";
             _scopeAttributes = scopeAttributes?.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
             RawResponse = rawResponse;
-            DefaultPollingInterval = TimeSpan.FromSeconds(1);
+            PollingStrategy = ChoosePollingStrategy(rawResponse, defaultPollingStrategy);
+        }
+
+        private static OperationPollingStrategy ChoosePollingStrategy(Response rawResponse, OperationPollingStrategy? defaultPollingStrategy)
+        {
+            if (RetryAfterPollingStrategy.TryParseAndBuild(rawResponse, out RetryAfterPollingStrategy? pollingStrategy))
```

Comment:
by delegating this logic we still have some coupling between the choosing and the polling strategies themselves

Reply:
I realized that after we move `ChoosePollingStrategy` to `OperationPollingStrategy` abstract class, there is no coupling anymore. `TryParseAndBuild` and `TryGetRetryAfter` are put on `OperationPollingStrategy`, and it's natural for `RetryAfterPolingStrategy` to use some codes in its base class.